### PR TITLE
Application configuration for production docker compose

### PR DESCRIPTION
**Pull request by thehabbos007**
## 🌟 What is the purpose of this PR?

After #1756 we've removed default URLs in the prod kratos config, which caused the production docker compose to fail as it did.

This PR fixes the issues found with the docker-compose not allowing sign-in. Transiently an error with the client-side vs. server-side apollo client was found and fixed.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203363157432094/1203672472334691/f) _(internal)_

## 🔍 What does this change?

- Adds env vars to configure kratos as it expects
- Fixes client-side vs. server-side API origin inconsistencies in the hash frontend
- Exposes prod mailslurper

## ❓ How to test this?

1. Make a .env.local with `HASH_SEED_USERS` set
1. Run the production docker compose `docker compose --file packages/hash/external-services/docker-compose.prod.yml --env-file .env.local up --build` 
1. Test out the application at http://localhost:3000/

### CODE REVIEW

File: packages/hash/external-services/docker-compose.prod.yml [hunk 1]
```diff
@@ -126,13 +137,12 @@ services:
 
   mailslurper:
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/external-services/docker-compose.prod.yml:138]: This was previously `read_only`, which caused mailslurper to fail immediately as it uses sqlite under the hood to manage state. If we want to make this `read_only` we have to mount a volume.

**Review by benwerner01 — COMMENTED** [review]
Changes look good to me, still trying to get the production docker-compose running locally. Had one clarifying question/suggestion which I will go ahead and leave now.

File: packages/hash/external-services/docker-compose.prod.yml [hunk 2]
```diff
@@ -126,13 +137,12 @@ services:
...
-    read_only: true
     security_opt:
       - no-new-privileges:true
     # These ports can be useful for introspection/debugging the mock mail server.
-    # ports:
-    #   - "4436:4436"
-    #   - "4437:4437"
+    ports:
+      - "4436:4436"
+      - "4437:4437"
```
[the remark this answers is not in the document]
Comment (thehabbos007) [reply packages/hash/external-services/docker-compose.prod.yml:145]: I think we don't want mailslurper in this prod docker compose but I also think it's a big ask to assume a working mailserver to run this in production mode. 

Kratos would work fine without (especially if we remove the `watch-courier` flag from the command)

**Review by benwerner01 — APPROVED** [review]

**Review by Alfred-Mountfield — APPROVED** [review]
Just tried it locally and it works, thanks

**Pull request merged**